tests/tools/oesign/test-enclave/host/host.c

```diff
@@ -20,12 +40,34 @@ int main(int argc, const char* argv[])
     }
 
     // Create the enclave
-    const uint32_t flags = oe_get_create_flags();
+    uint32_t flags = oe_get_create_flags();
+    bool is_kss_supported = _is_kss_supported();
+
+    /* Load the ELF image */
+    if ((result = oe_load_enclave_image(argv[1], &oeimage)) != OE_OK)
+    {
+        oe_put_err("oe_load_enclave_image(): result=%u", result);
+    }
+
+    /* Load the SGX enclave properties */
+    if ((result = oe_sgx_load_enclave_properties(
+             &oeimage, OE_INFO_SECTION_NAME, &properties)) != OE_OK)
+    {
+        oe_put_err("oe_sgx_load_enclave_properties(): result=%u", result);
+    }
 
     if ((result = oe_create_oesign_test_enclave(
```

Comment:
Continuing discussion from #3659:
> After I remove `_is_kss_supported` from oesign, test will fail here so I will update the test here like the following flow:
> ```c
> if ((result = oe_create_oesign_test_enclave(
>    argv[1], OE_ENCLAVE_TYPE_AUTO, flags, NULL, 0, &enclave)) != OE_OK)
> {
>     if (!(result == OE_PLATFORM_ERROR && is_kss_supported() && non-zero KSS properties))
>         oe_put_err("oe_create_crypto_enclave(): result=%u", result);
>     else
>     {
>         return 0;
>     }
> }
> ```

1. The proposed change is not reflected in this PR, so how are the tests passing now?
2. The proposed change is incorrect. It will cause all oesign tests, even those that do not rely on KSS properties to skip the enclave creation step, which is one of the necessary validations as part of those tests. 

My recommendation here remains unchanged:
> I think you're going to restructure this test, because once enclaves can be signed on non-KSS machines with KSS properties, the oe_create_enclave on line 45 will fail before it gets to this check, so you're going to have to introduce a mechanism to skip entire tests based on detection of this bit, and a parameter passed into the host to indicate whether a specific test expects KSS or not.
>
> See tests/debug-mode for an example.

To elaborate:
- The test host needs to take a command line argument indicating whether the test is intended to use KSS properties. 
   - This means updating the sign-and-verify.py and the tests CMakeLists.txt to specify and pass through this option to the test host.
   - If the option is provided and KSS is not supported, then the test host will skip the `oe_create_enclave` and `check_kss_extended_ids` verifications. 
      - The host should explicitly print a message like "Skipping creation of enclave using KSS on platform without KSS support..." (see similar message for MRSIGNER check in simulation mode).
      - This message is needed as an alternative success indicator, since all tests will look for the "PASS: Launch of signed enclave succeeded" set as part of `set_test_properties` in the tests' CMakeLists.txt. (i.e. needs to be provided as a second value delimited by `;` for [PASS_REGULAR_EXPRESSION](https://cmake.org/cmake/help/latest/prop_test/PASS_REGULAR_EXPRESSION.html)

Reply:
The new updated logic is as following:  Raise the exception in oe_sgx_build_enclave when context is OE_SGX_LOAD_TYPE_CREATE and platform doesn't support KSS and attribute has KSS enabled

    if (props.config.attributes & OE_SGX_FLAGS_KSS)
    {
        if ((context->type == OE_SGX_LOAD_TYPE_CREATE) && !_is_kss_supported())
        {
            // Fail if the CPU does not support KSS and the enclave specifies the KSS flag
            OE_RAISE_MSG(
                OE_UNSUPPORTED,
                "Enclave image was signed with kss flag but CPU doesn't support KSS\n",
                NULL);
        }
        context->attributes.flags |= OE_ENCLAVE_FLAG_SGX_KSS;
    }
At test tests\tools\oesign\test-enclave\host\host.c
Handle the OE_UNSUPPORTED error code and skip the test if attributes enable the KSS. The other tests will not be impacted. 
    if ((result = oe_create_oesign_test_enclave(
             argv[1], OE_ENCLAVE_TYPE_AUTO, flags, NULL, 0, &enclave)) != OE_OK)
    {
        if (!is_kss_supported &&
            (result == OE_UNSUPPORTED &&
             properties.config.attributes & OE_SGX_FLAGS_KSS))
        {
            // Skip the test as it is not supported
            return 0;
        }
        else
            oe_put_err("oe_create_crypto_enclave(): result=%u", result);
    }
Result for platform doesn't support :
38: Created /home/acctestuser/openenclave/build/tests/tools/oesign/test-enclave/enclave/oesign_test_enc.signed
38: 2020-11-03T23:20:30+0000.526527Z [(H)ERROR] tid(0x7f07702bbb80) | Enclave image was signed with kss flag but CPU doesn't support KSS
38:  (oe_result_t=OE_UNSUPPORTED) [/home/acctestuser/openenclave/host/sgx/create.c:oe_sgx_build_enclave:879]
38: 2020-11-03T23:20:30+0000.526657Z [(H)ERROR] tid(0x7f07702bbb80) | :OE_UNSUPPORTED [/home/acctestuser/openenclave/host/sgx/create.c:oe_create_enclave:1090]
1/1 Test #38: tests/oesign-sign-valid-kss ......   Passed    0.05 sec

Reply:
Thanks for the clarification, I agree that it works, but it still creates misleading test logs. 

- sign-and-verify.py still expects that the enclave launch passes when it invokes the test host, and so continues to report with a fixed message that enclave launch succeeded, when that's not the case.
- the script believes that is actually the case because the test host just eats the error and reports that the test passed completely with no logging that it skipped the test. 

I would suggest at least fixing the following:
- Update the default success message for `launch_cmd` in sign-and-verify.py to reflect the new meaning of a successful return from test host (e.g. "Signed enclave test app succeeded")
   - This does mean updating the CMakeList.txt files still, but it's a straightfoward global find and replace now
- Update test host to `printf` a message indicating that a KSS signed enclave load was skipped as unsupported by the platform (i.e. the comment on line 66 is no enough, we want that information to be indicated in the test results)

Reply:
Thanks. Updated per comments (update the default success message and test host to add printf message. )